File: src/Backend/Inventory.Data/Models/Sales/SalesInvoiceModel.cs
using Inventory.Data.Models.CRM;

namespace Inventory.Data.Models.Sales
{
    public class SalesInvoiceModel
    {
        /// <summary>
        /// The unique identifier for the sales invoice.
        /// </summary>
        public int InvoiceId { get; set; } = 0;

        /// <summary>
        /// Purchase date for this invoice.
        /// </summary>
        public string SalesDate { get; set; } = string.Empty;

        /// <summary>
        /// The total sales price for the invoice.
        /// </summary>
        public decimal InvoiceTotal { get; set; } = 0;

        /// <summary>
        /// The total amount of money paid to this vendor against this invoice.
        /// </summary>
        public decimal TotalPaid { get; set; } = 0;

        /// <summary>
        /// The total discount value on the invoice.
        /// </summary>
        public decimal TotalDiscount { get; set; } = 0;

        /// <summary>
        /// Current status of payment for this invoice
        /// </summary>
        public string PaymentStatus { get; set; } = RedBook.Core.Constants.CommonConstants.PaymentStatus.PartiallyPaid;

        /// <summary>
        /// The unique identifier for the organization associated with the sales invoice.
        /// </summary>
        public int OrganizationId { get; set; } = 0;

        /// <summary>
        /// The unique identifier for the sales invoice.
        /// </summary>
        public string InvoiceNumber { get; set; } = string.Empty;

        /// <summary>
        /// The remarks associated with the sales invoice.
        /// </summary>
        public string Remarks { get; set; } = string.Empty;

        /// <summary>
        /// The terms and conditions associated with the sales invoice.
        /// </summary>
        public string Terms { get; set; } = string.Empty;

        /// <summary>
        /// Person or entity, you are purchasing from.
        /// </summary>
        public CustomerModel? Customer { get; set; } = null;

        /// <summary>
        /// An array of sales details associated with the sales invoice.
        /// </summary>
        public List<SalesInvoiceDetailsModel> SalesDetails { get; set; } = new List<SalesInvoiceDetailsModel>();

        /// <summary>
        /// Payment records associated with the sales invoice.
        /// </summary>
        public List<SalesPaymentModel> PaymentRecords { get; set; } = new List<SalesPaymentModel>();
    }
}

File: src/Frontend/src/app/dashboard/Models/sales-invoice.model.ts
import { CustomerModel } from "./customer.model";
import { SalesInvoiceDetailsModel } from "./sales-invoice-details.model";
import { SalesPaymentModel } from "./sales-payment.model";

export class SalesInvoiceModel {
  /**
   * The unique identifier for the sales invoice.
   * @type {number}
   */
  invoiceId: number = 0;

  /**
   * Purchase date for this invoice.
   * @type {string}
   */
  salesDate: string = '';

  /**
   * The total sales price for the invoice.
   * @type {number}
   */
  invoiceTotal: number = 0;

  /**
   * The total amount of mmoney paid to this vendor against this invoice.
   * @type {number}
   */
  totalPaid: number = 0;

  /**
   * The total discount value on the invoice.
   * @type {number}
   */
  totalDiscount: number = 0;

  /**
   * Current status of payment for this invoice
   * @type {string}
   */
  paymentStatus: string = '';

  /**
   * The unique identifier for the organization associated with the sales invoice.
   * @type {number}
   */
  organizationId: number = 0;

  /**
   * The unique identifier for the sales invoice.
   * @type {string}
   */
  invoiceNumber: string = '';

  /**
   * The remarks associated with the sales invoice.
   * @type {string}
   */
  remarks: string = '';

  /**
   * The terms and conditions associated with the sales invoice.
   * @type {string}
   */
  terms: string = '';

  /**
   * Person or entity name of the customer
   * @type {string}
   */
  customerName: string = 'Guest User';

  /**
   * Person or entity details object of the customer
   * @type {string}
   */
  customer: CustomerModel | null = null;

  /**
   * An array of sales details associated with the sales invoice.
   * @type {Array<SalesInvoiceDetailsModel>}
   */
  salesDetails: Array<SalesInvoiceDetailsModel> = [];

  /**
   * Payment records associated with the sales invoice.
   * @type {Array<SalesPaymentModel>}
   */
  paymentRecords: Array<SalesPaymentModel> = [];
}
